core/src/main/scala/akka/persistence/cassandra/query/EventsByTagStage.scala

```diff
@@ -301,9 +301,10 @@ import com.datastax.driver.core.utils.UUIDs
           tagPidSequenceNrs += (repr.persistenceId -> ((1, repr.offset)))
           push(out, repr)
           false
-        } else if (usingOffset && currTimeBucket.inPast) {
+        } else if (usingOffset && (currTimeBucket.inPast || settings.eventsByTagNewPersistenceIdScanTimeout == scala.concurrent.duration.Duration.Zero)) {
```

Comment:
import scala.concurrent.duration.Duration

Reply:
Doh, we had a Duration from the Cassandra driver in scope as well